### DISCUSSION TITLE
Support horizontal scroll in ScrollRestoration

### DISCUSSION THREAD

**Discussion by alekcey0211** [remix-run/react-router#11997] [Proposals]
### What version of React Router are you using?

6.26.0

### Steps to Reproduce

I have a report with many columns and a horizontal scroll. I store the column sort direction in the search parameters. When I change the last column's sort direction, the page scrolls back to the start.

### Expected Behavior

I'm expecting that horizontal scroll position restores too

### Actual Behavior

Horizontal scroll position always reset to 0

**Comment by brophdawg11** [comment]
`ScrollRestoration` is only designed to handle to vertical scroll at the moment.  I'm going to convert this to a a Proposal since it's a Feature request.  In the meantime you can also look at https://github.com/epicweb-dev/restore-scroll as an example of how you can build your own more advanced scroll restoration on top of the React Router APIs.
